doc/naming.md

```diff
@@ -14,33 +14,43 @@ be plugged in.
 ### Name Syntax
 
 A fully qualified, self contained name used for gRPC channel construction
-uses the syntax:
+uses URI syntax as defined in [RFC 3986](https://tools.ietf.org/html/rfc3986).
 
-```
-scheme://authority/endpoint_name
-```
+The URI scheme indicates what resolver plugin to use.  If no scheme
+prefix is specified or the scheme is unknown, the `dns` scheme is used
+by default.
 
-Here, `scheme` indicates the name-system to be used. Currently, we
-support the following schemes:
+The URI path indicates the name to be resolved.
 
-- `dns`
+Most gRPC implementations support the following URI schemes:
 
-- `ipv4` (IPv4 address)
+- `dns:[//authority/]host[:port]` -- DNS (default)
+  - `host` is the host to resolve via DNS.
+  - `port` is the port to return for each address.  If not specified,
+    443 is used.
```

Comment:
Java does 80 if plaintext and 443 if secure.

Reply:
Go always uses 443.

Also, for Go, "`host`" is optional.  We allow "`:80`" and assume `localhost` in that case.

Reply:
Added note about some implementations using 80 for insecure channels.

Having "host" be optional is very strange, and I don't think that's something we should be supporting.